.github/workflows/checks.yaml

```diff
@@ -0,0 +1,374 @@
+  #         ./zpipe_no_std < Cargo.toml | ./zpipe_no_std -d > out.txt
+  #         cmp -s Cargo.toml out.txt
+  #     - name: "cdylib: custom-prefix"
+  #       working-directory: libz-rs-sys-cdylib
+  #       env:
+  #         LIBZ_RS_SYS_PREFIX: "MY_CUSTOM_PREFIX_"
+  #       run: |
+  #         cargo build --release --target ${{matrix.target}} --features=custom-prefix
+  #         objdump -tT target/${{matrix.target}}/release/deps/libz_rs.so | grep -q "MY_CUSTOM_PREFIX_uncompress" || (echo "symbol not found!" && exit 1)
+
+  wasm32:
+    name: "wasm32"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+        with:
+          persist-credentials: false
+          submodules: true
+      - name: Install wasmtime
+        run: curl https://wasmtime.dev/install.sh -sSf | bash
+      - name: Copy .cargo/config.toml
+        run: |
+          mkdir -p .cargo
+          cp qemu-cargo-config.toml .cargo/config.toml
+      - name: Install toolchain
+        uses: dtolnay/rust-toolchain@be73d7920c329f220ce78e0234b8f96b7ae60248
+        with:
+          toolchain: "stable"
+          targets: "wasm32-wasip1"
+      - name: target
+        run: "rustc -vV | sed -n 's|host: ||p'"
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@56ab7930c591507f833cbaed864d201386d518a8
+        with:
+          tool: cargo-nextest
+      - name: Download wasi-sdk
+        run: |
+          curl -L https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-24/wasi-sdk-24.0-x86_64-linux.tar.gz | tar xz
+      - name: cargo nextest (without SIMD)
+        run: |
+          export CC="$(pwd)/wasi-sdk-24.0-x86_64-linux/bin/clang"
+          export CFLAGS="-target wasm32-wasip1"
+          cargo nextest run -p libbzip2-rs-sys -p test-libbzip2-rs-sys --target wasm32-wasip1 \
+            -- --skip high_level_write --skip open_and_close # These tests use temp_dir
+        env:
+          RUST_BACKTRACE: 1
+          RUSTFLAGS: ""
+
+  miri:
+    name: "Miri"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+        with:
+          persist-credentials: false
+          submodules: true
+      - name: Install Miri
+        run: |
+          rustup toolchain install nightly --component miri
+          cargo +nightly miri setup
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@56ab7930c591507f833cbaed864d201386d518a8
+        with:
+          tool: cargo-nextest
+      - name: Test public C api with NULL arguments
+        run: "cargo +nightly miri nextest run -j4 -p test-libbzip2-rs-sys null::"
+        env:
+          RUSTFLAGS: "-Ctarget-feature=+avx2"
+      - name: Test allocator with miri
+        run: "cargo +nightly miri nextest run -j4 -p libbzip2-rs-sys allocate::"
```

Comment:
this matches no tests in our case right? we should make a subset somehow that can run in < 1min

Reply:
Right...